controller/users.js

```diff
@@ -1,14 +1,78 @@
-exports.createUser = async (ctx, next) => {
-  try {
-  } catch (error) {}
-};
+const bcrypt = require('bcrypt');
+const atob = require('atob');
+const jwt = require('jsonwebtoken');
 
-exports.signIn = async (ctx, next) => {
+const { user: User } = require('../db/models');
+const { filterProps } = require('../utils/filterProps');
+
+exports.createUser = async ctx => {
   try {
-  } catch (error) {}
+    const userData = ctx.request.body;
+    let user = await User.findOne({ where: { email: userData.email } });
+    if (user) {
+      ctx.status = 400;
+      ctx.body = {
+        error: 'User already exists',
+      };
+    } else {
+      if (userData.password) {
+        const hash = await bcrypt.hash(userData.password, 10);
+        userData.password = hash;
+      }
+      user = await User.create({ ...userData });
+      user = filterProps(user.dataValues, [
+        'id',
+        'picture',
+        'password',
```

Comment:
Don't send back to the client the password, this expose important information in the client.

Reply:
The elements in the array state, which entries should be removed from the object. Hence password is currently not send back to the client. Let's discuss tomorrow.

Reply:
I see, the name was misunderstanding since i thought was keeping those props. Ok cool, kn this case it works :)